extensions/npm/src/tasks.ts

```diff
@@ -298,50 +308,53 @@ export function getTaskName(script: string, relativePath: string | undefined) {
 	return script;
 }
 
-export async function createTask(packageManager: string, script: INpmTaskDefinition | string, cmd: string[], folder: WorkspaceFolder, packageJsonUri: Uri, scriptValue?: string, matcher?: any): Promise<Task> {
-	let kind: INpmTaskDefinition;
-	if (typeof script === 'string') {
-		kind = { type: 'npm', script: script };
-	} else {
-		kind = script;
-	}
-
-	function getCommandLine(cmd: string[]): (string | ShellQuotedString)[] {
-		const result: (string | ShellQuotedString)[] = new Array(cmd.length);
-		for (let i = 0; i < cmd.length; i++) {
-			if (/\s/.test(cmd[i])) {
-				result[i] = { value: cmd[i], quoting: cmd[i].includes('--') ? ShellQuoting.Weak : ShellQuoting.Strong };
-			} else {
-				result[i] = cmd[i];
-			}
-		}
-		if (workspace.getConfiguration('npm', folder.uri).get<boolean>('runSilent')) {
-			result.unshift('--silent');
+function escapeCommandLine(cmd: string[]): (string | ShellQuotedString)[] {
+	const result: (string | ShellQuotedString)[] = new Array(cmd.length);
+	for (let i = 0; i < cmd.length; i++) {
+		if (/\s/.test(cmd[i])) {
+			result[i] = { value: cmd[i], quoting: cmd[i].includes('--') ? ShellQuoting.Weak : ShellQuoting.Strong };
+		} else {
+			result[i] = cmd[i];
 		}
-		return result;
 	}
+	return result;
+}
+
+function getRelativePath(rootUri: Uri, packageJsonUri: Uri): string {
+	const absolutePath = packageJsonUri.path.substring(0, packageJsonUri.path.length - 'package.json'.length);
+	return absolutePath.substring(rootUri.path.length + 1);
+}
 
-	function getRelativePath(packageJsonUri: Uri): string {
-		const rootUri = folder.uri;
-		const absolutePath = packageJsonUri.path.substring(0, packageJsonUri.path.length - 'package.json'.length);
-		return absolutePath.substring(rootUri.path.length + 1);
+export async function getRunScriptCommand(script: string, folder: Uri, context?: ExtensionContext, showWarning = true): Promise<string[]> {
+	const scriptRunner = await getScriptRunner(folder, context, showWarning);
+	const result = [scriptRunner, scriptRunner === 'node' ? '--run' : 'run'];
+	if (workspace.getConfiguration('npm', folder).get<boolean>('runSilent')) {
+		result.push('--silent');
```

Comment:
`--silent` will cause an error if the scriptRunner is node, we should ignore that setting in this case